services/security/authentication/casigned/casigned.go

```diff
@@ -37,6 +44,47 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 	return &response.Token, nil
 }
 
+func RenewRoutine(group, server string) {
+	renewalAttempt := 0
+	// Waiting for a few seconds to avoid spamming short-lived sdk user
+	time.Sleep(time.Second * 5)
+	for {
+		wssdConfig := auth.WssdConfig{}
+		err := marshal.FromJSONFile(auth.GetWssdConfigLocation(), &wssdConfig)
+		if err != nil {
+			fmt.Printf("Failed to open config file in location %s: %v\n", auth.GetWssdConfigLocation(), err)
```

Comment:
+1 To what Jocelyn says.
Don't panic and don't use fmt.Printf.  
Return the error you are printing out and have the caller handle it.

Reply:
The caller is unaware of the renewal process, it is in a background thread that the renewal is happening